ocaml/lang/security/no-md5.yaml

```diff
@@ -0,0 +1,16 @@
+rules:
+- id: no-md5
+  patterns:
+  - pattern-either:
+    - pattern: |
+        Hash.md5
+    - pattern: |
+        MAC.hmac_md5
```

Comment:
IIRC HMAC-MD5 has no known practical vulnerabilities yet:

- https://crypto.stackexchange.com/questions/9336/is-hmac-md5-considered-secure-for-authenticating-encrypted-data
- https://en.wikipedia.org/wiki/HMAC#Security

Best practices would still avoid HMAC-MD5, but it's not broken. So, depending on what we're going for here, we may want to rethink including this pattern.

Reply:
good point, we should only catch md5 for now